src/wp-includes/class-wp-theme.php

```diff
@@ -374,13 +374,26 @@ public function __construct( $theme_dir, $theme_root, $_child = null ) {
 		}
 
 		// If we got our data from cache, we can assume that 'template' is pointing to the right place.
-		if ( ! is_array( $cache ) && $this->template != $this->stylesheet && ! file_exists( $this->theme_root . '/' . $this->template . '/index.php' ) ) {
+		if (
+			! is_array( $cache ) &&
+			$this->template != $this->stylesheet &&
+			! file_exists( $this->theme_root . '/' . $this->template . '/index.php' ) &&
+			! file_exists( $this->theme_root . '/' . $this->template . '/templates/index.html' ) &&
+			! file_exists( $this->theme_root . '/' . $this->template . '/block-templates/index.html' ) // Deprecated path support since 5.9.0.
+		) {
 			// If we're in a directory of themes inside /themes, look for the parent nearby.
 			// wp-content/themes/directory-of-themes/*
 			$parent_dir  = dirname( $this->stylesheet );
 			$directories = search_theme_directories();
 
-			if ( '.' !== $parent_dir && file_exists( $this->theme_root . '/' . $parent_dir . '/' . $this->template . '/index.php' ) ) {
+			if (
+				'.' !== $parent_dir &&
+				(
+					file_exists( $this->theme_root . '/' . $parent_dir . '/' . $this->template . '/index.php' ) ||
+					file_exists( $this->theme_root . '/' . $parent_dir . '/' . $this->template . '/templates/index.html' ) ||
+					file_exists( $this->theme_root . '/' . $parent_dir . '/' . $this->template . '/block-templates/index.html' ) // Deprecated path support since 5.9.0.
```

Comment:
is using `is_readable` a better check here?

Reply:
Looking at [is_block_theme](https://github.com/WordPress/wordpress-develop/blob/acbbee8a1199f9ce70f1d3bbcc9cb80e32047e6c/src/wp-includes/class-wp-theme.php#L1493) it uses `is_file() && is_readable()`. I don't know why `is_readable()` is really needed - if the file exists, even if we can't access it, it's a block theme.. `file_exists()` seems the appropriate (and probably technically faster) function.